packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1723,16 +1722,13 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         this.dataStores.processSignal(envelope.address, transformed, local);
     }
 
-    public async getRootDataStore(id: string, wait = true): Promise<IFluidRouter> {
-        return this.getRootDataStoreChannel(id, wait);
-    }
-
-    private async getRootDataStoreChannel(id: string, wait = true): Promise<IFluidDataStoreChannel> {
+    public async getRootDataStore(id: string, wait = true): Promise<IDataStore> {
         await this.dataStores.waitIfPendingAlias(id);
         const internalId = this.internalId(id);
         const context = await this.dataStores.getDataStore(internalId, wait);
         assert(await context.isRoot(), 0x12b /* "did not get root data store" */);
-        return context.realize();
+        const channel = await context.realize();
+        return channelToDataStore(channel, id, this, this.dataStores, this.mc.logger);
```

Comment:
this will also expose trySetAlias on retrieved root data stores. I don't think this is something we want

Reply:
Is there a reason for that? I did realize it would, but it felt like making our interfaces to a data store more consistent might be a good thing, instead of getting objects with different functionality on a "getDataStore" vs a "createDataStore".

I ran into some issue with this discrepancy when I did an experiment merging main into the old branch and thus decided to make this change, but maybe I can flip the inheritance in a way that this doesn't happen and still avoids the problem I ran into, I'll look into that option.

Reply:
Ok, and in fact to not expose it I just need to cast `as IFluidDataStoreRuntimeEntrypoint` here, no need to flip anything in the inheritance chain. Still curious if there's a reason why root data stores should not be alias-able?

Reply:
fyi @andre4i 

Aliasing is a create time operation. it gives the data store a strong name, and prevents it from being garbage collected, it should be an uncommon operation. There are still open questions in this area, but we shouldn't expose new functionality like this without careful though and design. 


There really is no get data store. We special case root which pre-dates and is being replaced by aliases. We will need a mechanism for the container runtime itself to get root/alias datastores, but ideally the scope of that is as minimal as possible, and even in that case we shouldn't allow re-aliasing those datastores.

Reply:
I think with the changes you are makeing IDataStore will implement IFluidHandle, and so will IFluidDataStoreChannel. since DataStore, the implementation of IDataStore wraps the underlying IFluidDataStoreChannel, it should just delegate to handle that will be exposed off IFluidDataStoreChannel

Reply:
in general, i would probably go with an inside out strategy for building this, rather than an outside in. it's good to have clarity on what the outside api will look like, but it should be possible to start transparently building up from the inside, and then light up the outside api as a last step. 

Also, i imagine in this first wave of changes that the handle will be optional at all layers, and we'll make it required at a later date.

Reply:
> if there's a reason why root data stores should not be alias-able?

Aliasing is intended to replace the root datastore creation logic. Creating root datastores has been deprecated and the only way of creating a 'rooty' datastore is by having a regular datastore and aliasing it. The reason the `IDataStore` interface exists is so that clients can only alias regular datastores. Aliasing was designed to fix a frequent name conflict which happened at root datastore creation and corrupted the container, it not a way to redirect/rename datastores...